### PR TITLE
Implement interceptor which parse collection operators and stores res…

### PR DESCRIPTION
**Pull request by ystankevich**
…ult in context

### CODE REVIEW

**Comment by ystankevich** [comment]
That allow in future, to modify collection operators  for developers, store them in context and underlying code will use that new collection operators.

File: gateway/middleware.go [hunk 1]
```diff
@@ -29,143 +34,104 @@ func UnaryServerInterceptor() grpc.UnaryServerInterceptor {
...
+//
+// Collection operators setters
+//
+func CtxSetFiltering(ctx context.Context, f *query.Filtering) context.Context {
+	return context.WithValue(ctx, filteringCollection, f)
+}
 
-	for i := 0; i < reqval.NumField(); i++ {
-		f := reqval.FieldByIndex([]int{i})
+func CtxSetSorting(ctx context.Context, s *query.Sorting) context.Context {
```
Comment (Evgeniy-L) [inline gateway/middleware.go]: Please add proper comments to methods.

**Comment by dmacthedestroyer** [comment]
What's the intention for this PR? It looks like you're trying to store the collection operators in context to be able to modify it through the middleware chain. 

Does this make it so that the collection operators in the protobuf request message can become out of sync with what's in the context?

**Comment by ystankevich** [comment]
That is fix for NGP-1019.
In that fix implemented functions which allow to get collection parameters from context.
Also allow to create new context with new collection parameters.
So developer may modify or use specific collection parameters even if they are not provided by initial request

**Comment by ystankevich** [comment]
We will partly use that for supporting filtering by synthetic fields.

**Comment by dmacthedestroyer** [comment]
My main concern is that if someone puts a collection operator in their request message proto, there's a possibility that because of this middleware the value provided to their service implementation will be different from what's being stored in the context. 

Is there any assurance that these two places won't get out of sync?

**Pull request closed without merging**